src/libraries/System.Drawing.Common/src/System/Drawing/Printing/PrinterSettings.cs

```diff
@@ -1141,17 +1148,18 @@ internal PrinterResolution[] Get_PrinterResolutions()
         }
 
         // names is pointer to DEVNAMES
-        private static string ReadOneDEVNAME(IntPtr pDevnames, int slot)
+        private static unsafe string ReadOneDEVNAME(IntPtr pDevnames, int slot)
         {
-            int offset = checked(Marshal.SystemDefaultCharSize * Marshal.ReadInt16((IntPtr)(checked((long)pDevnames + slot * 2))));
-            string result = Marshal.PtrToStringAuto((IntPtr)(checked((long)pDevnames + offset)))!;
+            byte* bDevNames = (byte*)pDevnames;
+            int offset = checked(Marshal.SystemDefaultCharSize * Marshal.ReadInt16((IntPtr)(bDevNames + slot * 2)));
```

Comment:
```suggestion
            int offset = Marshal.SystemDefaultCharSize * ((ushort*)bDevNames)[slot];
```
This does not need to be checked